pkg/bench/get.go

```diff
@@ -47,6 +51,56 @@ type Get struct {
 // Prepare will create an empty bucket or delete any content already there
 // and upload a number of objects.
 func (g *Get) Prepare(ctx context.Context) error {
+
+	/*
+	 * Prepare the bench by listing object from the bucket
+	 */
+	if g.GetFromListing {
+		cl, done := g.Client()
+
+		// ensure the bucket exist
+		found, err := cl.BucketExists(ctx, g.Bucket)
+		if err != nil {
+			return err
+		}
+		if !found {
+			return(fmt.Errorf("bucket %s does not exist and --get-from-bucket-listing has been set", g.Bucket))
+		}
+
+		// list all objects
+		objectCh := cl.ListObjects(ctx, g.Bucket, minio.ListObjectsOptions{
+			Prefix: g.ListingPrefix,
+			Recursive: g.ListingRecursive,
+		})
+		for object := range objectCh {
+			if object.Err != nil {
+				return object.Err
+			}
+			if object.Size == 0 {
+				continue
+			}
+			obj := generator.Object {
+				Name: object.Key,
+				Size: object.Size,
+			}
+			g.objects = append(g.objects, obj)
+
+			// limit to ListingMaxObjects
+			if g.ListingMaxObjects > 0 && len(g.objects) >= g.ListingMaxObjects {
+				break
```

Comment:
Provide a cancelable ctx to `ListObjects` and cancel it when we return, so it isn't leaking resources while we run the benchmark.

Reply:
fixed